File: EXAMEN/Problema1.py
import datetime
import psycopg2


conexion = psycopg2.connect(
    host = 'localhost',
    port = '5432',
    user = 'postgres',
    password = 'admin',
    dbname = 'postgres'
)
cursor = conexion.cursor()
try:
    print("Ingrese los datos ")
    dia = int(input("dia de nacimiento: "))
    mes = int(input("mes de nacimiento: "))
    año = int(input("Año de nacimiento: "))

except:
    print("Los datos son erroneos, ingrese numeros")

año_act = datetime.date.today().year
mes_act = datetime.date.today().month
dia_act = datetime.date.today().day

fecha_act = dia_act+mes_act+año_act
cumple = dia+mes+año_act

if (fecha_act>cumple):
    edad_años = (año_act -año)
else:
    edad_años = ((año_act)-año)-1

edad_mes = abs(mes_act-mes)
edad_dia = abs(dia_act-dia)

print("La persona cumple: ",edad_años)

if (fecha_act)>=(cumple):
    print("La persona ya cumplió años")
else : 
    print("La persona aun no ha cumplido años")

cursor.execute("Insert into problema1 VALUES (%s, %s, %s, %s);",(dia,mes,año,edad_años))

cursor.close()
conexion.commit()
conexion.close()

File: EXAMEN/Problema1.m
clc; clear;
try
  pkg load database
  conn = pq_connect(setdbopts('dbname','postgres','host','localhost','port','5432','user','postgres','password','admin'))
  e1 = input('Ingrese dia de nacimiento: ');
  e2 = input('Ingrese mes de nacimiento: ');
  e3 = input('Ingrese año de nacimiento: ');
  
if (2022>e3 && e3>1900)
    edad = 2022-e3;
  if (e2<=32 && e2>=1)
    if (e2<=31 && e1>=1)
      if (e2<=8 && e1<=26)
        printf('Su edad es %d, usted ya cumplió años',edad)
        else
        printf('Su edad es %d, usted no ha cumplido años',edad)
        end
      else 
      printf('Fecha incorrect')
        end
    else
    printf('Fecha incorrecta')
    end
  else 
printf('Fecha incorrecta')
pq_exec_params(conn, 'insert into problema1 values ($1,$2,$3);',{e1,e2,e3});
end   
end